spring-cloud-gcp-starters/spring-cloud-gcp-starter-core/src/main/java/org/springframework/cloud/gcp/core/autoconfig/GcpContextAutoConfiguration.java

```diff
@@ -33,25 +39,40 @@
  * Binds properties from {@link GcpProperties}
  *
  * @author Vinicius Carvalho
+ * @author João André Martins
  */
 @Configuration
 @ConditionalOnClass(GoogleCredentials.class)
 @EnableConfigurationProperties(GcpProperties.class)
 public class GcpContextAutoConfiguration {
 
+	private static final String GCP_PUBSUB_SCOPE = "https://www.googleapis.com/auth/pubsub";
+
 	@Autowired
 	private GcpProperties gcpProperties;
 
 	@Bean
 	@ConditionalOnMissingBean
-	public GoogleCredentials googleCredentials() throws Exception {
+	public CredentialsProvider googleCredentials() throws Exception {
 		if (!StringUtils.isEmpty(this.gcpProperties.getCredentialsLocation())) {
-			return GoogleCredentials.fromStream(
-					this.gcpProperties.getCredentialsLocation().getInputStream());
-		}
-		else {
-			return GoogleCredentials.getApplicationDefault();
+			return FixedCredentialsProvider.create(GoogleCredentials.fromStream(
+					this.gcpProperties.getCredentialsLocation().getInputStream()));
 		}
+
+		return GoogleCredentialsProvider.newBuilder()
+				.setScopesToApply(ImmutableList.of(GCP_PUBSUB_SCOPE))
```

Comment:
`Collections.singletonList()` is much better. Really optimized for a single value and is *immutable* as well.